### DISCUSSION TITLE
Is it possible to fit a DataTable to the container?

### DISCUSSION THREAD

**Discussion by sauljabin** [Textualize/textual#1942] [Q&A] [answered]
Hello folks,

First, I want to congratulate you all, you are doing a great job, 

I'm developing the version 2 of kaskade https://github.com/sauljabin/kaskade/tree/v2 (please visit it, and let me know your feedback). I have drastically reduced the amount of code using the newer textual release, and that is really great.

I'm eager to have Tabs and TextAreas (I'm waiting for them, for a couple of new functionalities I'll add to kaskade).

I bring a question related to DataTables, I would like to know if it is possible to set an attribute that allows a DataTable to fit to the container, I have this:

![](https://raw.githubusercontent.com/sauljabin/kaskade/v2/screenshots/new-dashboard1.png)

But with bigger windows it looks like this:

![](https://raw.githubusercontent.com/sauljabin/kaskade/v2/screenshots/new-dashboard2.png)

**Comment by willmcgugan** [comment, marked as answer]
Looks great! Love that you have tweaked the colors.

You might want to add the number cells as `Text(str(number), justify="right")`

We will have flexible columns at some point. And the column header bar will stretch to the full width in the next update.

Tabs and Textareas are coming too!

**Reply by sauljabin** [reply]
Nice advice, definitely I do, I'll try `justify="right"` (is it possible through css? it's a great feature https://github.com/sauljabin/kaskade/blob/v2/kaskade/tui.css).

Thanks, I will wait for flexible columns so (and tabs and textareas).

**Comment by valerio-afk** [comment]
I needed the same thing, more or less.
I wanted to have columns that width are meant as percentage of the size of the DataTable and I made this workaround

```
    def post_display_hook(this) -> None:
        table = this.query_one("#mydatatable")
        size = table.size

        if all([c for c in size]):

            for c in table.columns.values():
                c.auto_width=False
                if (not hasattr(c,"percentage_width") or (c.percentage_width is None)):
                    c.percentage_width = c.width

                c.width = int(this.size[0] * (c.percentage_width / 100))

        table.refresh()
```

I am adding a new property to the object DataTable. I achieve it by adding this line of code in the __init__ of my App

`Column.percentage_width = None `

such that I don't change the original API

Then I work out the current size of the DataTable and take percentages of the width.

I know it's not the best way to code it, but it's a workaround that is doing its job for what I need.
